### PR TITLE
Initial implementation of Linux support.

### PR DESCRIPTION
**Pull request by gedaskir**
This is initial implementation of MIKE IO 1D working on Linux. The main changes are related to deployment of the package:

- When release is triggered two virtual machine instances are started, which build platform specific wheels, i.e.
    -  **manylinux2010_x86_64** (note that we do not perform **auditwheel** and only a rename from **linux_x86_64** to **manylinux2010_x86_64** is done).
    -  **win_amd64**
- Dependency on **DHI.MikeCore.Linux.rhel7 20.0.0** NuGet package is added for Linux.
- The **runtimeconfig.json**  is instructed to use latest available .NET version in the system by using `"rollForward": "LatestMajor"` option.

There are a few things, which I don't think are fully proper, but still I think it is a good idea to already merge it in into main:

- For example, the **mikecore-python** package dependency should not be necessary. I use it here because it helped to resolve the native libraries correctly, because **mikecore-python** is deployed with _patchelfed_ libraries. To my understanding setting LD_LIBRARY_PATH in Python process environment does not really help as per this thread:
https://stackoverflow.com/questions/23244418/set-ld-library-path-before-importing-in-python
For now we have MIKE Core binaries in two places.

- ~~Another thing is the strict requirement of .NET Core 3.1 presence. In principle it should work with any .NET 5 and above.~~

- Having 10 checks in GitHub actions maybe is a bit too much. Additionally, we get Python.NET not closing properly once in a while and failing tests (some garbage collector failure).

### CODE REVIEW

**Comment by ecomodeller** [comment]
> This is exciting :) ... I really look forward to being able to use mikeio1d with linux soon. It'll simplify installing some of the optional dependencies (like geopandas), as well as allow things like devcontainers.
> 
> These are my main comments:
> 
> * Should we add MacOS at the same time? I also wondered about testing a few other linux distros, but it seems github actions only includes ubuntu as convenience.
> * I think we should include some high-level end-to-end tests that show expected results are returned regardless of the OS. These could compare all timeseries results for a few different files to what is expected. We could mark these as 'slow' tests that are only run when merging into main.

I don't think that <https://www.nuget.org/packages/DHI.MikeCore.Linux.rhel7> will work on Mac OS, but I am happy to be proven wrong🙂.

File: .github/workflows/full_test.yml [hunk 1]
```diff
@@ -26,7 +27,7 @@ jobs:
     - name: Set up .NET
       uses: actions/setup-dotnet@v3
       with:
-        dotnet-version: '5.0'
+        dotnet-version: '3.1.x'
```
Comment (ecomodeller) [inline .github/workflows/full_test.yml:30]: .NET 3.1 reached end-of-life in 2022😳

Comment (gedaskir) [reply .github/workflows/full_test.yml]: @ecomodeller I updated the runtimeconfig.json to use latest available .NET runtime and setup-dotnet@v3 to use .NET 6.0. Apparently was quite simple :-).

**Comment by ecomodeller** [comment]
💖🚀

I can review the code in a codespace. Well done!
image

**Comment by gedaskir** [comment]
@rywm-dhi The newly introduced test_core.py fails now, because of the slight differences in the date time indices (for example, **'1994-08-07 16:36:01.869000'** vs. **'1994-08-07 16:36:01.870000'**). It is not fully clear if it is a rounding, different MIKE Core version (on Linux we use version 20.0, because that is the only available in NuGet now), running Linux vs. Windows, or etc. issue.

File: README.md
[the remark this answers is not in the document]
Comment (gedaskir) [reply README.md:1]: This is a good idea.

**Pull request merged**